test/snippet/CMakeLists.txt

```diff
@@ -51,7 +51,10 @@ macro (seqan3_snippet target_cpp snippet_full_path)
 
     add_executable (${target} ${source_path}/${target_cpp})
     target_link_libraries (${target} seqan3::test::unit)
-    # target_compile_options(${target} PUBLIC --disable-werror)
+    set_target_properties(${target}
+        PROPERTIES
+        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${source_path}"
```

Comment:
Not sure about the source structure, but: `${CMAKE_BINARY_DIR}` is probably what you are looking for?
It is the mirrored structure of the cmake sources in the binary directory.